packages/web-components/fast-foundation/src/di/di.ts

```diff
@@ -1513,26 +1487,40 @@ export class ContainerImpl implements Container {
     }
 
     constructor(protected owner: any, protected config: ContainerConfiguration) {
-        if (owner !== null) {
-            owner.$$container$$ = this;
-        }
-
         this.resolvers = new Map();
         this.resolvers.set(Container, containerResolver);
 
-        if (owner instanceof Node) {
-            owner.addEventListener(
-                DILocateParentEventType,
-                (e: CustomEvent<DOMParentLocatorEventDetail>) => {
-                    if (e.composedPath()[0] !== this.owner) {
-                        e.detail.container = this;
+        if (owner) {
+            (owner as any).$$container$$ = this;
+
+            if ("addEventListener" in owner) {
+                Context.handle(owner, (e: ContextEvent<UnknownContext>) => {
+                    if (this.isHandlingContextRequests) {
+                        try {
+                            const value = this.get(e.context);
+                            e.stopImmediatePropagation();
+                            e.callback(value);
+                        } catch {
+                            // Container failed to find the context, so we need to
+                            // let the event propagate.
+                            // TODO: Introduce a tryGet API to Container.
```

Comment:
Should this be tracked as an issue?

Reply:
Actually, we've got an issue for that. I'll add the issue number.